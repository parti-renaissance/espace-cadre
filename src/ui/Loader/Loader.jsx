import { CircularProgress, Box } from '@mui/material'
import PropTypes from 'prop-types'

const Loader = ({ size = 16, color = 'main', isCenter = false }) => {
  const loader = <CircularProgress size={size} sx={{ color }} />

  if (isCenter) {
    return (
      <Box sx={{ width: '100%', height: '20vh', display: 'flex', alignItems: 'center', justifyContent: 'center' }}>
        {loader}
      </Box>
    )
  }

  return loader
}

Loader.propTypes = {
  size: PropTypes.number,
  color: PropTypes.string,
  isCenter: PropTypes.bool,
}

export default Loader
