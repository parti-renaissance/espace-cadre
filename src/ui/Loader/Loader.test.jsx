import { render } from '@testing-library/react'
import Loader from 'ui/Loader/Loader'

vi.mock('@mui/material', () => ({
  CircularProgress: ({ size }) => <div>Mock CircularProgress: size {size}</div>,
}))

describe('Loader', () => {
  it('displays Loader', () => {
    const { container } = render(<Loader />)

    expect(container).toMatchSnapshot()
  })
})
