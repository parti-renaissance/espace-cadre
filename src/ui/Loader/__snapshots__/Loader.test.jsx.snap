// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loader > displays Loader 1`] = `
<div>
  <div>
    Mock CircularProgress: size 
    16
  </div>
</div>
`;
